### DISCUSSION TITLE
Converting between types

### DISCUSSION THREAD

**Discussion by andrewragy** [typst/typst#1214] [Q&A] [answered]
How does one convert between types in typst?
E.g. is there a function that takes 12 and converts it to "12" (integer to string) or vice versa?

**Comment by laurmaedje** [comment, marked as answer]
`str(12)` converts to a string and `int("12")` converts to an integer

**Reply by andrewragy** [reply]
Ok, thank you

**Comment by admercs** [comment]
How to convert a `str` or `int` to `relative length` type or `length` type? I want to do math on the length types like so:

```typst
#let result1 = length(1).cm - 1cm
#let result2 = 1 - 1cm
```

The compiler should be able to infer the latter.
